File: tests/fixtures/dependency-injection-pos/impl.js
// SMELL: the class constructs its own collaborator internally, so it cannot be
// tested or reconfigured without reaching into globals. Candidate for DI.
class IdGen {
  next() { return "id-fixed"; }
}

class OrderService {
  constructor() {
    // hard-wired collaborator
    this.ids = new IdGen();
    this.created = [];
  }
  create(name) {
    const id = this.ids.next();
    this.created.push({ id, name });
    return id;
  }
}

module.exports = { OrderService };

File: tests/fixtures/py-dependency-injection-pos/impl.py
# SMELL: the class constructs its own collaborator internally, so it cannot be
# tested or reconfigured without reaching into globals. Candidate for DI.


class IdGen:
    def next(self) -> str:
        return "id-fixed"


class OrderService:
    def __init__(self):
        # hard-wired collaborator
        self._ids = IdGen()
        self.created = []

    def create(self, name: str) -> str:
        new_id = self._ids.next()
        self.created.append({"id": new_id, "name": name})
        return new_id
